tests/unittest.py

```diff
@@ -150,7 +151,11 @@ def deepcopy_config(config: _TConfig) -> _TConfig:
     return new_config
 
 
-_make_homeserver_config_obj_cache: Dict[str, Union[RootConfig, Config]] = {}
+@functools.lru_cache(maxsize=8)
```

Comment:
Is there anyway to know how many of these actually get re-used?

Reply:
There are actually 359 distinct configs used in tests, not 80 as I had written.
With a size of 8, the LRU cache reports `CacheInfo(hits=1676, misses=380, maxsize=8, currsize=8)`.
With an unlimited size, the LRU cache reports `CacheInfo(hits=1697, misses=359, maxsize=None, currsize=359)`.

The histogram of config usage looks like
1249, 57, 33, 30, 20, 20, 19, 16, 14, 13, 12, 12, 11, 11, 11, 10, 9, 8,
8, 8, 8, 7, 6, 6, 6, 6, 6, 5, 5, 5, 5, 5, 4, 4, 4, 4, 4, 4, 4, 4, 3, 3,
3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 2, 2, 2, 2, 2, 2, 2, 2, 2,
2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1, 1, 1,
... followed by a long tail of ones.

There's one main config that's used by most tests and then some others.

Reply:
Seems like `8` is a reasonable size then. 😄